IFTB.bs

```diff
@@ -0,0 +1,535 @@
+
+The <code>featureRecord</code> array must be sorted by <code>featureTag</code> with any feature tag
+occurring at most once.  Only layout features that have their own chunks should be have a
+<code>FeatureRecord</code> <code>firstIndex</code> is the chunk index of the first chunk specific to
+this feature. The number of feature-specific chunks will then be <code>chunkRecordCount</code> which
+must be at least 1. The chunk correspondence is encoded in <code>ChunkMapRecords</code>
+
+```
+chunkIdx      firstChunk
+chunkIdx      lastChunk
+```
+
+The <code>chunkMapRecord</code> array contains as many entries as the sum of <code>chunkMapCount</code>
+fields in the <code>featureRecord</code> array, with <code>chunkMapRecord[0]</code> corresponding to
+the first chunk of <code>featureRecord[0]</code>,
+<code>chunkMapRecord[featureRecord[0].chunkMapCount]</code> corresponding to the first chunk of
+<code>featureRecord[1]</code>, <code>chunkMapRecord[featureRecord[0].chunkMapCount +
+chunkMapRecord[featureRecord[1].chunkMapCount]</code> corresponding to the first chunk of
+<code>featureRecord[2]</code>, and so on.
+
+<code>firstChunk</code> must be less than or equal to <code>lastChunk</code>. If <code>featureTag</code>
+is part of the set of layout features in the font subset description, and any of firstChunk through
+lastChunk are required by the set of codepoints in that description, then that feature-specfic chunk
+must also be loaded.
+
+### Optional IFTB Table Values ### {#iftb-table-optional}
+
+If a particular IFTB-encoded font does not include a range file the chunkRangeFileURI string must be
+empty. The <code>chunkOffsetList</code> can either be included for reference or omitted by setting the
+<code>chunkOffsetListOffset</code> field to 0.  If a particular font does not include separate chunk
+files the <code>chunkFilesURI</code> field must be empty.
+
+When an IFTB format font has no feature-specific chunks the <code>featureMapOffset</code> must be 0.
+
+IFTB Chunk Formats {#iftb-chunk-formats}
+----------------------------------------
+
+### The Uncompressed Chunk Format (IFTC) ### {#iftc-format}
+
+An uncompressed chunk string, which is normally produced by decoding a compressed chunk, has these
+fields:
+
+```
+uint32        version     - must be "IFTC" tag
+uint32        reserved    - must be 0
+uint32        id[4]       - unique id -- must match same fields in IFTB table
+uint32        chunkIndex  - (uint32) for potential future extensions
+uint32        length      - the length of the (uncompressed) chunk string
+uint32        glyphCount  - the number of glyphs encoded in the chunk
+uint8         tableCount  - the number of tables the chunk has data for (must be 1 or 2)
+uint16        GIDs[glyphCount]    - an array of GIDs included in the chunk
+uint32        tables[tableCount]  - an array of tables (by tag) included in the chunk.
+Offset32      offsets[glyphCount * tableCount] - an array of offsets of glyph data,
```

Comment:
Instead of 32bit offsets could this be 16 bit lengths? There's no reason that a client would need random access to the pointed to glyph data. Like gids this array could use a variable width int type (8, 16, or 32 bit depending on what's needed to accommodate the particular lengths in the chunk).